osu.Game.Rulesets.Osu/Skinning/IHasSpinnerApproachCircle.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+
+namespace osu.Game.Rulesets.Osu.Skinning
+{
+    /// <summary>
+    /// A common interface between implementations of the <see cref="OsuSkinComponents.SpinnerBody"/> component that provide approach circles for the spinner.
+    /// </summary>
+    public interface IHasSpinnerApproachCircle
```

Comment:
I'm not sure this needs to be a spinner-specific interface?

Reply:
Could make it `IHasApproachCircle`, just felt it doesn't need to go outside of its scope if it's only used for the spinner.